.github/workflows/_ios-build-test.yml

```diff
@@ -67,27 +67,20 @@ jobs:
           # Install dependencies
           brew install libtool
 
-      - name: Install conda and dependencies
-        run: |
-          # Install conda, setup-miniconda messes with the path that messes with the ruby stuff we do later on
-          curl --retry 3 --retry-all-errors -o "${RUNNER_TEMP}/conda.sh" https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
-          chmod +x "${RUNNER_TEMP}/conda.sh"
-          /bin/bash "${RUNNER_TEMP}/conda.sh" -b -p "${RUNNER_TEMP}/anaconda"
-          echo "${RUNNER_TEMP}/anaconda/bin" >> "${GITHUB_PATH}"
-          # shellcheck disable=SC1091
-          source "${RUNNER_TEMP}/anaconda/bin/activate"
-          conda install -y \
-            blas=1.0 \
-            cffi=1.15.1 \
-            cmake=3.22.1 \
-            mkl=2022.1.0 \
-            mkl-include=2022.1.0 \
-            ninja=1.10.2 \
-            numpy=1.23.3 \
-            pyyaml=6.0 \
-            requests=2.28.1 \
-            setuptools=63.4.1 \
-            typing_extensions=4.3.0
+      - name: Setup miniconda for iOS device
+        if: inputs.ios-platform != 'SIMULATOR'
+        uses: pytorch/test-infra/.github/actions/setup-miniconda@main
+        with:
+          python-version: "3.10"
+          environment-file: .github/requirements/conda-env-iOS
+
+      - name: Setup miniconda for iOS simulator
+        if: inputs.ios-platform == 'SIMULATOR'
+        uses: pytorch/test-infra/.github/actions/setup-miniconda@main
+        with:
+          python-version: "3.10"
+          environment-file: .github/requirements/conda-env-iOS
```

Comment:
Is this possible to use the following syntax https://github.com/pytorch/pytorch/blob/master/.github/workflows/_mac-test.yml#L104-L105 here

```
environment-file: .github/requirements/conda-env-${{ runner.os }}-${{ runner.arch }}
pip-requirements-file: .github/requirements/pip-requirements-${{ runner.os }}.txt
```

It it works (all ios and macos tests pass), you can reuse the env and requirements files for macos and just need to add more dependencies there if needed instead of creating two news files.  Let's try it out

Reply:
Switched to using a common conda environment for iOS and MacOS. Both tests pass.

However, I've run into issues when using a common pip requirements file (`FileNotFoundError: [Errno 2] No such file or directory: 'llvm-config'` which may be related `numba` required by MacOS). All tests pass if use a distinct requirements file for iOS simulator.

Reply:
I have rebased the PR to make sure that the PR is on top of the latest stable commit.  The macos x86-64 failures https://github.com/pytorch/pytorch/actions/runs/3780917647/jobs/6427614261 is a bit of a concern, so I want to see if it's still persist after rebasing and rerunning the test.  If it goes away, all is good.  If the failure persists, I think your previous approach of using a separate conda file is the correct one (as I suspect updating macos x86-64 conda file has an unexpected side effect causing that test to fail)

Reply:
MacOS test failure seems to persist. I've reverted to using a separate iOS file for conda requirements.